sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -433,6 +433,18 @@ class SessionCatalog(
   }
 
   /**
+   * Return whether a table with the specified name is a data source table.
+   */
+  def isDatasourceTable(name: TableIdentifier): Boolean = synchronized {
```

Comment:
Yeah, `DDLUtils` is in `/sql/core`. We are unable to call them in `/sql/catalyst`. I am thinking we should move them into `/sql/catalyst`. I hit another issue in https://github.com/apache/spark/pull/13070. Do you think it is OK?

BTW, after following  your suggestion, we do not need it in this PR. Will remove it.